src/main/java/controller/Controller.java

```diff
@@ -0,0 +1,69 @@
+package controller;
+
+import domain.*;
+import view.InputView;
+import view.OutputView;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class Controller {
+    private InputView inputView;
+    private OutputView outputView;
+    private BooleanGenerator booleanGenerator;
+
+    public Controller(InputView inputView, OutputView outputView, BooleanGenerator booleanGenerator) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.booleanGenerator = booleanGenerator;
+    }
+
+    public void run() {
+        List<String> playerNames = inputView.readPlayerName();
+        Players players = new Players(generatePlayers(playerNames));
+
+        int ladderHeight = inputView.readLadderHeight();
+        List<Line> ladder = generateLadder(ladderHeight, players);
+
+        printResult(players, ladder);
+    }
+
+    private List<Player> generatePlayers(List<String> playerNames) {
+        List<Player> players = new ArrayList<>();
+        for (String playerName : playerNames) {
+            players.add(new Player(playerName));
+        }
+        return players;
+    }
+
+    private List<Line> generateLadder(int ladderHeight, Players players) {
+        return IntStream.range(0, ladderHeight)
+                .mapToObj(i -> generateLine(players))
+                .collect(Collectors.toList());
+    }
+
+    private Line generateLine(Players players) {
+        Block preBlock = new Block(booleanGenerator.generate());
+        List<Block> blocks = createBlocks(players, preBlock);
+        return new Line(players, blocks);
+    }
+
+    private List<Block> createBlocks(Players players, Block preBlock) {
+        List<Block> blocks = new ArrayList<>(List.of(preBlock));
+
+        IntStream.range(1, players.getPlayersSize() - 1)
+                .mapToObj(i -> new Block(booleanGenerator.generate()))
+                .forEach(nextBlock -> {
+                    nextBlock.comparePreBlock(preBlock);
+                    blocks.add(nextBlock);
+                });
```

Comment:
사실상 2중포문인데 메서드 분리가 되어있지 않아 가독성이 2중포문보다 나쁘다고 느껴졌습니다! forEach는 Consumer를 받는데요, Consumer를 반환하는 메서드를 만들면 메서드 분리가 가능합니다.
```suggestion
                .forEach(blockLineChange(preBlock, blocks));
```
혹은 인자를 받는 void 메서드를 만들수도 있고요.
```suggestion
                .forEach(nextBlock -> changeBlockLine(blocks, preBlock, nextBlock));
```

그러나 근본적으로.. 전 이 로직에서 IntStream 자체가 주는 이점이 없다고 느낍니다. 
1. 연산이 많아 선언형 스트림의 가독성을 느낄수 있는 것도 아니고, 
2. 쇼트 서킷, 지연처리 등의 스트림 내장 기능의 혜택을 받을 수 있는 것도 아니며, (단일 블록에 forEach 처리)
3. 스트림 생성 비용으로 인한 성능적 손해를 보고 있기 때문입니다.

이 부분에 대한 아코의 생각을 들려주세용~~! (야호~ 개발토론이다)

Reply:
저도 일단 스트림이 비용이 높아서 코딩테스트나 시간복잡도가 중요한 사항에서는 사용하지 않는다고 접했습니다.  그럼에도 제가 이번 미션에서 Stream을 과하게 사용해본 이유는 미션을 통해 stream을 활용하는 것을 연습해보고 싶어서 사용해보았습니다. 웨지 말대로 처음보는 사람 입장에서는 코드가 가독성이 높지 않다라는 것을 인지하게 되었고 for문을 이용하든 혹은 stream 이용하든 가독성이 높은 코드를 작성하는 것이 중요하다는 것을 배울 수 있었습니다.

Reply:
후후 개발토론 열리자마자 닫혀서 아쉽군요 
아코의 생각을 들려주어서 감사합니다!! 학습을 위한 사용도 중요하다고 생각합니다.